metrics/ci_integrations/lib/client/buildkite/mappers/buildkite_build_state_mapper.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:ci_integration/client/buildkite/models/buildkite_build_state.dart';
+import 'package:ci_integration/integration/interface/base/client/mapper/mapper.dart';
+
+/// A class that provides methods for mapping Buildkite build statuses.
+class BuildkiteBuildStateMapper implements Mapper<String, BuildkiteBuildState> {
+  /// A running build status.
+  static const String running = 'running';
+
+  /// A scheduled build status.
+  static const String scheduled = 'scheduled';
+
+  /// A successful finished build status.
+  static const String passed = 'passed';
+
+  /// A failed build status.
+  static const String failed = 'failed';
+
+  /// A status that indicates the build blocked by another build.
```

Comment:
It seems like the blocked build's state doesn't stand for a build that is blocked by another build. Please check the [block step](https://buildkite.com/docs/pipelines/block-step) for a pipeline. I suppose we should test this and check when the build has the state `blocked`. Also, calling the [list builds endpoint](https://buildkite.com/docs/apis/rest-api/builds#list-builds-for-a-pipeline) with the `state` query parameter with `finished` value lists blocked builds as well as passed, failed, and canceled 🤔